Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs

```diff
@@ -507,12 +507,32 @@ public ActionResult StatsChart(int contestId)
                     .Select(ContestProblemListViewModel.FromProblem),
             };
 
+            var participantsToSkip = 0;
+            var participantsToTake = int.MaxValue;
+
+            if (page.HasValue)
+            {
+                participantsToSkip = (page.Value - 1) *
+                                     (official
+                                         ? OfficialResultsPageSize
+                                         : NotOfficialResultsPageSize);
+
+                participantsToTake = official
+                    ? OfficialResultsPageSize
+                    : NotOfficialResultsPageSize;
+            }
+
             var participants = this.participantsData
                 .GetAllByContestAndIsOfficial(contest.Id, official)
-                .ToList()
-                .OrderByDescending(p => p.TotalScoreSnapshot);
+                .OrderByDescending(p => p.TotalScoreSnapshot)
+                .Skip(participantsToSkip)
```

Comment:
ThenBy(p => p.TotalScoreSnapshotModifiedOn). We want the users who completed the contest first, to appear first in the results.

Reply:
Done